### PR TITLE
Enhanced doris streamx connector

### PR DESCRIPTION
**Pull request by monrg**
### What problem does this PR solve?

Enhanced doris streamx connector

## Purpose of this pull request
 This pull request adds auto parse doris  properties ， add async flush value to doris, can controll max row ... threshold ， Enhanced doris streamx connector

### CODE REVIEW

File: streamx-flink/streamx-flink-connector/streamx-flink-connector-doris/src/main/java/com/streamxhub/streamx/flink/connector/doris/bean/LoadStstusFailedException.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.streamxhub.streamx.flink.connector.doris.bean;
+
+import java.io.IOException;
+import java.util.Map;
+
+public class LoadStstusFailedException extends IOException {
```
Comment (lvshaokang) [inline streamx-flink/streamx-flink-connector/streamx-flink-connector-doris/src/main/java/com/streamxhub/streamx/flink/connector/doris/bean/LoadStstusFailedException.java]: May be should LoadStatusFailedException?

Comment (wolfboys) [reply streamx-flink/streamx-flink-connector/streamx-flink-connector-doris/src/main/java/com/streamxhub/streamx/flink/connector/doris/bean/LoadStstusFailedException.java]: > May be should LoadStatusFailedException?

+1 😂

Comment (monrg) [reply streamx-flink/streamx-flink-connector/streamx-flink-connector-doris/src/main/java/com/streamxhub/streamx/flink/connector/doris/bean/LoadStstusFailedException.java]: i fix it

File: streamx-flink/streamx-flink-connector/streamx-flink-connector-doris/src/main/java/com/streamxhub/streamx/flink/connector/doris/internal/DorisSinkFunction.java [hunk 2]
```diff
@@ -19,167 +19,120 @@
...
     @Override
-    public void initializeState(FunctionInitializationContext functionInitializationContext) throws Exception {
+    public void initializeState(FunctionInitializationContext context) throws Exception {
+        if (Semantic.EXACTLY_ONCE.equals(Semantic.of(dorisConfig.semantic()))) {
+            ListStateDescriptor<Map<String, DorisSinkBufferEntry>> descriptor =
+                new ListStateDescriptor<>(
+                    "buffered-rows",
+                    TypeInformation.of(new TypeHint<Map<String, DorisSinkBufferEntry>>() {
+                    })
+                );
```
Comment (JNSimba) [inline streamx-flink/streamx-flink-connector/streamx-flink-connector-doris/src/main/java/com/streamxhub/streamx/flink/connector/doris/internal/DorisSinkFunction.java:125]: thanks for your pr,
Store data in state. When there is a lot of data and the checkpoint time is long, the state may be particularly large?
I think it's better to implement exactly-once based on [flink](https://flink.apache.org/features/2018/03/01/end-to-end-exactly-once-apache-flink.html) and [doris's](https://doris.apache.org/zh-CN/docs/data-operate/import/import-way/stream-load-manual.html) two-phase commit, what do you think?

Comment (wolfboys) [reply streamx-flink/streamx-flink-connector/streamx-flink-connector-doris/src/main/java/com/streamxhub/streamx/flink/connector/doris/internal/DorisSinkFunction.java:125]: > thanks for your pr, Store data in state. When there is a lot of data and the checkpoint time is long, the state may be particularly large? I think it's better to implement exactly-once based on [flink](https://flink.apache.org/features/2018/03/01/end-to-end-exactly-once-apache-flink.html) and [doris's](https://doris.apache.org/zh-CN/docs/data-operate/import/import-way/stream-load-manual.html) two-phase commit, what do you think?

 +1

**Pull request merged**